### PR TITLE
Add Nuget Version support in Univers

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
TODO:
Add ABOUT file for Nuget code
Add NugetVersion Range and add tests for it

File: src/univers/nuget.py.ABOUT [hunk 1]
```diff
@@ -0,0 +1,11 @@
+about_resource: maven.py
```
Comment (pombredanne) [inline src/univers/nuget.py.ABOUT:1]: I am not sure you meant maven?

File: src/univers/nuget.py.ABOUT [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+package_url: pkg:github/google/osv
+copyright: |
+    Copyright 2022 Google LLC
+
+notes: This has been substantially modified and enhanced from the original 
+ pymaven code to extract the version comparison code.
+
```
Comment (pombredanne) [inline src/univers/nuget.py.ABOUT:8]: We need a direct download URL including a commit hash

File: src/univers/nuget.py.NOTICE [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2022 Google LLC
```
Comment (pombredanne) [inline src/univers/nuget.py.NOTICE:1]: We likely want this also in the nuget.py.

File: src/univers/nuget.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
+#
+# Copyright (c) nexB Inc. and others.
```
Comment (pombredanne) [inline src/univers/nuget.py:2]: Is this rather copyright Google? You probably want an initial commit with the original unmodified files? And did you modify this code?
I see that https://github.com/google/osv/blob/f5647ad2f746685b08debfba0293e442f2fb9945/lib/osv/semver_index.py#L1 and https://github.com/google/osv/blob/f5647ad2f746685b08debfba0293e442f2fb9945/lib/osv/nuget.py both have a notice and we surely do not want to remove their copyrights, even if this is still in the ABOUT and NOTICE file

File: src/univers/nuget.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+# Visit https://aboutcode.org and https://github.com/nexB/univers for support and download.
+
+import functools
+import re
+
+import semver
```
Comment (pombredanne) [inline src/univers/nuget.py:10]: Could we eventually avoid having two semver libraries and reuse `semantic-version` we already use here?
 @rbarrois https://pypi.org/project/semantic-version/ https://github.com/rbarrois/python-semanticversion that we use can parse node-semver version specs constraints while https://github.com/python-semver/python-semver/ can only deal with versions and not the npm specs (we use to convert from npm native to our vers syntax)  which is why we picked in the first place.

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks! ... here are some nits for your consideration!

File: src/univers/nuget.py.ABOUT (hunk 1 above)
Comment (TG1999) [reply src/univers/nuget.py.ABOUT:1]: Oops, fixing this

File: src/univers/nuget.py (hunk 5 above)
Comment (TG1999) [reply src/univers/nuget.py:10]: I tried reusing the existing library at first place, It returns prerelease and build as tuples, whereas the newly imported one returns a string, and there is no concrete way to get to strings from those tuples.
For example:
```
semantic_version.Version.coerce("1.0.0.0").prerelease
('0',)
nuget.Version.from_string("1.0.0.0").prerelease
None
```
That's why I think we should use the new library for nuget

Comment (TG1999) [reply src/univers/nuget.py:18]: One more thing I observed here, the equal comparison of both the libraries are different, I tried to run all the tests after making the required adjustments by using the `semantic_version` library, but some corner cases are failing there, one more reason to use the new library for nuget

Comment (TG1999) [reply src/univers/nuget.py:18]: Here is the gist for same https://gist.github.com/TG1999/724c868fa21882aa880c091cf5e1964d

Comment (TG1999) [reply src/univers/nuget.py:18]: "1.0.0", "1.0.0+beta"

Comment (pombredanne) [reply src/univers/nuget.py:18]: > "1.0.0", "1.0.0+beta"

So if I get you, nuget consider that these tow are equal? https://github.com/nexB/univers/pull/60/files#diff-1b35636a23b0127a5ed3b726fe14efdf9df1a159e27dd87ead2be71547b78cccR39

This is mind blowing!

*sigh* ok, well, so let's use this for now. :|

File: tests/test_nuget.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Google LLC
+#
```
Comment (pombredanne) [inline tests/test_nuget.py:2]: Adding the missing copyright from upstream https://github.com/NuGet/NuGet.Client/blob/eea4636696cc7b6a3e682797c3dd717ad25fd7c9/test/NuGet.Core.Tests/NuGet.Versioning.Test/VersionComparerTests.cs
```suggestion
# Copyright (c) .NET Foundation. All rights reserved.
```

**Review by pombredanne — APPROVED** [review]
LGTM! Thanks!
